### PR TITLE
Fix java time epoch date formatters

### PR DESCRIPTION
**Pull request by spinscale**
The self written epoch date formatters were not properly able to format
an Instant to a string due to a misconfiguration.

This fix also removes a until now existing runtime behaviour under java
8 regarding the names of the aggregation buckets, which are now the same
as before and have been under java 11.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/common/time/EpochTime.java [hunk 1]
```diff
@@ -72,7 +70,7 @@ public TemporalAccessor resolve(Map<TemporalField,Long> fieldValues,
     private static final EpochField NANOS_OF_SECOND = new EpochField(ChronoUnit.NANOS, ChronoUnit.SECONDS, ValueRange.of(0, 999_999_999)) {
         @Override
         public boolean isSupportedBy(TemporalAccessor temporal) {
-            return temporal.isSupported(ChronoField.NANO_OF_SECOND) && temporal.getLong(ChronoField.NANO_OF_SECOND) != 0;
+            return temporal.isSupported(ChronoField.NANO_OF_SECOND);
```
Comment (rjernst) [inline server/src/main/java/org/elasticsearch/common/time/EpochTime.java]: I added this condition to guard from printing out ".0" when nanos are 0. Why is this being removed? This test passes currently but fails with this PR:

```
DateFormatter secondsFormatter = DateFormatter.forPattern("epoch_second");
Instant instant = Instant.ofEpochSecond(42, 0);
assertThat(secondsFormatter.format(instant), equalTo("42"));
```

Comment (spinscale) [reply server/src/main/java/org/elasticsearch/common/time/EpochTime.java]: hm, this one passed for me on java8 and java11 in this PR. Any test setup that I am missing?

Comment (spinscale) [reply server/src/main/java/org/elasticsearch/common/time/EpochTime.java]: the trick is, that you can have an optional fractional with a length of `0`

**Comment by spinscale** [comment]
@elasticmachine retest elasticsearch-ci/1

**Review by rjernst — APPROVED** [review]
LGTM. Thanks for figuring out the trick about zero length fraction! The test I ran before passes now.

**Pull request merged**